sites/en/job-board/create_a_rails_app.step

```diff
@@ -20,23 +20,108 @@ message <<-MARKDOWN
   # Rails New!!!
 MARKDOWN
 
-console "rails new job_board -T --skip-turbolinks"
+option "Without Docker" do
+  console "rails new job_board -T --skip-turbolinks"
+
+  message <<-MARKDOWN
+    The `-T` in that command means that when you make new files using Rails generators, it doesn't automatically create test files using Test::Unit (the default Rails testing framework).
+
+    Watch all the files that are created! Wow!
+  MARKDOWN
+
+  console_with_message "Move into the directory for your new Rails app:", "cd job_board"
+end
+
+option "With Docker" do
+  console "mkdir job_board"
+  console "cd job_board"
+
+  message "In your editor, create a new file called `Gemfile`"
+  message "Copy the following into `Gemfile`:"
+  source_code :ruby, <<-CONTENTS
+    source "https://rubygems.org"
+    gem "rails", "5.2.0"
+  CONTENTS
+  console "touch Gemfile.lock"
+
+  message "In your editor, create a new file called `Dockerfile`:"
+  message "Copy the following into `Dockerfile`:"
+  source_code :text, <<-CONTENTS
+    FROM ruby:2.5.1-slim
+
+    RUN apt-get update -qq && apt-get install -y build-essential libpq-dev git vim nodejs postgresql-client build-essential
+
+    ENV APP_HOME /job_board
+    RUN mkdir -p $APP_HOME
+    WORKDIR $APP_HOME
+    COPY Gemfile* $APP_HOME/
+    RUN bundle install --binstubs
```

Comment:
Remove the " --binstubs" portion otherwise after following the job_board instructions if you run a `bin/rails` command you get the following very verbose help output mixed with the output of the actual command you ran (e.g. generate a controller).

```
job_board: (master) docker-compose run web rails generate controller jobs
Starting job_board_db_1 ... done
Beginning in Rails 4, Rails ships with a `rails` binstub at ./bin/rails that
should be used instead of the Bundler-generated `rails` binstub.

If you are seeing this message, your binstub at ./bin/rails was generated by
Bundler instead of Rails.

You might need to regenerate your `rails` binstub locally and add it to source
control:

 rails app:update:bin           # Bear in mind this generates other binstubs
                                # too that you may or may not want (like yarn)

If you already have Rails binstubs in source control, you might be
inadverently overwriting them during deployment by using bundle install
with the --binstubs option.

If your application was created prior to Rails 4, here's how to upgrade:

  bundle config --delete bin    # Turn off Bundler's stub generator
  rails app:update:bin          # Use the new Rails executables
  git add bin                   # Add bin/ to source control

You may need to remove bin/ from your .gitignore as well.

When you install a gem whose executable you want to use in your app,
generate it and add it to source control:

  bundle binstubs some-gem-name
  git add bin/new-executable

      create  app/controllers/jobs_controller.rb
      invoke  erb
      create    app/views/jobs
      invoke  helper
      create    app/helpers/jobs_helper.rb
      invoke  assets
      invoke    coffee
      create      app/assets/javascripts/jobs.coffee
      invoke    scss
      create      app/assets/stylesheets/jobs.scss
Usage:
  rails new APP_PATH [options]

Options:
      [--skip-namespace], [--no-skip-namespace]            # Skip namespace (affects only isolated applications)
  -r, [--ruby=PATH]                                        # Path to the Ruby binary of your choice
                                                           # Default: /usr/local/bin/ruby
  -m, [--template=TEMPLATE]                                # Path to some application template (can be a filesystem path or URL)
  -d, [--database=DATABASE]                                # Preconfigure for selected database (options: mysql/postgresql/sqlite3/oracle/frontbase/ibm_db/sqlserver/jdbcmysql/jdbcsqlite3/jdbcpostgresql/jdbc)
                                                           # Default: sqlite3
      [--skip-yarn], [--no-skip-yarn]                      # Don't use Yarn for managing JavaScript dependencies
      [--skip-gemfile], [--no-skip-gemfile]                # Don't create a Gemfile
  -G, [--skip-git], [--no-skip-git]                        # Skip .gitignore file
      [--skip-keeps], [--no-skip-keeps]                    # Skip source control .keep files
  -M, [--skip-action-mailer], [--no-skip-action-mailer]    # Skip Action Mailer files
  -O, [--skip-active-record], [--no-skip-active-record]    # Skip Active Record files
      [--skip-active-storage], [--no-skip-active-storage]  # Skip Active Storage files
  -P, [--skip-puma], [--no-skip-puma]                      # Skip Puma related files
  -C, [--skip-action-cable], [--no-skip-action-cable]      # Skip Action Cable files
  -S, [--skip-sprockets], [--no-skip-sprockets]            # Skip Sprockets files
      [--skip-spring], [--no-skip-spring]                  # Don't install Spring application preloader
      [--skip-listen], [--no-skip-listen]                  # Don't generate configuration that depends on the listen gem
      [--skip-coffee], [--no-skip-coffee]                  # Don't use CoffeeScript
  -J, [--skip-javascript], [--no-skip-javascript]          # Skip JavaScript files
      [--skip-turbolinks], [--no-skip-turbolinks]          # Skip turbolinks gem
  -T, [--skip-test], [--no-skip-test]                      # Skip test files
      [--skip-system-test], [--no-skip-system-test]        # Skip system test files
      [--skip-bootsnap], [--no-skip-bootsnap]              # Skip bootsnap gem
      [--dev], [--no-dev]                                  # Setup the application with Gemfile pointing to your Rails checkout
      [--edge], [--no-edge]                                # Setup the application with Gemfile pointing to Rails repository
      [--rc=RC]                                            # Path to file containing extra configuration options for rails command
      [--no-rc], [--no-no-rc]                              # Skip loading of extra configuration options from .railsrc file
      [--api], [--no-api]                                  # Preconfigure smaller stack for API only apps
  -B, [--skip-bundle], [--no-skip-bundle]                  # Don't run bundle install
      [--webpack=WEBPACK]                                  # Preconfigure for app-like JavaScript with Webpack (options: react/vue/angular/elm/stimulus)

Runtime options:
  -f, [--force]                    # Overwrite files that already exist
  -p, [--pretend], [--no-pretend]  # Run but do not make any changes
  -q, [--quiet], [--no-quiet]      # Suppress status output
  -s, [--skip], [--no-skip]        # Skip files that already exist

Rails options:
  -h, [--help], [--no-help]        # Show this help message and quit
  -v, [--version], [--no-version]  # Show Rails version number and quit

Description:
    The 'rails new' command creates a new Rails application with a default
    directory structure and configuration at the path you specify.

    You can specify extra command-line arguments to be used every time
    'rails new' runs in the .railsrc configuration file in your home directory.

    Note that the arguments specified in the .railsrc file don't affect the
    defaults values shown above in this help message.

Example:
    rails new ~/Code/Ruby/weblog

    This generates a skeletal Rails installation in ~/Code/Ruby/weblog.
```

Reply:
@javierjulio will defer to @maxjacobson and @pdarden since they are most familiar with " --binstubs"

Reply:
I think we're safe to remove the `--binstubs` flag from the `Dockerfile`. That flag tells bundler to generate files under the `bin` directory so you can do things like `bin/rails` instead of `bundle exec rails`. AFAIK our docs don't encourage writing `bin/rails` or `bin/rake` at any point, so we're not dependent on those files existing

Reply:
We do have use of `bin/rails` in command instructions that we need to correct. Are we good with dropping the binstubs flag and then updating command examples to drop `bin/`?

Reply:
We do have use of `bin/rails` in command instructions that we need to correct. Are we good with dropping the binstubs flag and then updating command examples to drop `bin/`? It would help to have someone follow through and confirm they get the same output as I pasted above following the JobBoard instructions.

Reply:
Hm. Another option would be to change the line to:

```
RUN bundle install --binstubs && rails app:update:bin
```

That might be worthwhile as it's a smaller change, I think would avoid the (omg so long) warning and also not require changing other things. WDYT?

Reply:
@maxjacobson sure, that would be fine but would that really work? At that point there is no rails app so no binstub files to update. Or would it?

Reply:
Good point, it might not work, on the first run, at which point rails is installed but the current working directory isn't (yet) a rails app. the `rails` binary might squawk about being in a non-rails-app context. It might be worth trying ...  if it doesn't work, then updating the references to `bin/rails` to be just `rails` (if that's a reasonably-small change) is probably the move